PhysicsTools/PatAlgos/python/slimming/prunedGenParticles_cfi.py

```diff
@@ -15,6 +15,7 @@
         "drop   status == 2 && abs(pdgId) == 21",                          # but remove again gluons in the inheritance chain
         "keep abs(pdgId) == 23 || abs(pdgId) == 24 || abs(pdgId) == 25 || abs(pdgId) == 6 || abs(pdgId) == 37 ",   # keep VIP(articles)s
         "keep abs(pdgId) == 310 && abs(eta) < 2.5 && pt > 1 ",                                                     # keep K0
+        "+keep abs(pdgId) == 13 && status == 1", # keep muon parents
```

Comment:
I think the reason why the ++keep of line 7 is not enough is that the kaons that decay to muons probably have status 2, and so get dropped at line 9.
for a very similar reason we have some more keeps in lines 11 and 12 for electron mothers and for taus.